6.4 Base Release notes (#3803)
astronomer/docs: astro/release-notes.md

<!-- ALL LINKS TO INTERNAL DOCS MUST BE COMPLETE URLS INCLUDING HTTPS. Otherwise the links will break in RSS. -->

## June 6, 2024

### API keys are no longer supported

As of June 1, 2024, Deployment API keys are no longer supported. Replace your API keys with [Deployment API tokens](https://www.astronomer.io/docs/astro/deployment-api-tokens), confirm successful operation with your API tokens, and then delete your API keys.

> ### Configurable ephemeral storage is now in Public Preview

Reviewer: Are we fast following with changes to core docs?

Author: Yes in #3804, there should be a link somewhere to them in this PR!

<HostedBadge/>

:::publicpreview
:::

You can now customize the size of the available ephemeral storage for data intensive workloads on your Celery and Kubernetes workers. Previously, to accomodate larger workloads, you might need to integrate with other object storage or database options to accomodate large datasets. Now, you can customize the ephemeral storage in the Astro UI when creating or updating your Deployment.

> - **Celery worker**: 10 GiB minimum by default. 100 GiB maximum.
> - **Kubernetes executor/ Kubernetes pod operator**: 0.25 GiB minimum by default. 100 GiB maximum.
> 
> You are only charged for resources used that are greater than the minimum defaults.

Reviewer: ```suggestion
You are only charged for used resources which are greater than the minimum defaults for each worker type:

- **Celery worker**: 10 GiB minimum by default. 100 GiB maximum.
- **Kubernetes executor/ Kubernetes pod operator**: 0.25 GiB minimum by default. 100 GiB maximum.

```

### Additional improvements

> - When you create a new Deployment in dev mode with hibernation mode enabled, Astro now validates CRON expressions that you enter and adds commonly used hibernation schedules that you can enable or disable. <!--https://github.com/astronomer/astro/pull/21337-->

Reviewer: ```suggestion
- The Astro UI now validates cron expressions for development Deployment hibernation schedules and shows commonly used hibernation schedules that you can enable or disable. <!--https://github.com/astronomer/astro/pull/21337-->
```
Is this really only for new Deployments? I'd be surprised if it was. I know that's the literal text in the issue but it warrants additional validation.

Reviewer: Also would be good to link the hibernation docs here. And in those docs we have an example cron schedule or two that can be used - should those now be removed because they're available in the UI?

Author: I'm working on those updates right now - was waiting for additional information that I just got this morning :)

> - You can now create a GitHub repository when you authorize the GitHub Integration from the Astro UI, if you do not already have an existing repository to connect to. Previously, you could only connect Astro to existing repositories. See [Deploy code with GitHub](https://www.astronomer.io/docs/astro/deploy-github-integration).

Reviewer: ```suggestion
- If you don't already have a GitHub repository, you can now create one when you authorize the [GitHub Integration](https://www.astronomer.io/docs/astro/deploy-github-integration) from the Astro UI. Previously, you could only connect Astro to existing repositories.
```

<!-- ALL LINKS TO INTERNAL DOCS MUST BE COMPLETE URLS INCLUDING HTTPS. Otherwise the links will break in RSS. -->

## June 6, 2024

### API keys are no longer supported

As of June 1, 2024, Deployment API keys are no longer supported. Replace your API keys with [Deployment API tokens](https://www.astronomer.io/docs/astro/deployment-api-tokens), confirm successful operation with your API tokens, and then delete your API keys.

### Configure ephemeral storage on worker Pods

<HostedBadge/>

:::publicpreview
:::

You can now customize the amount of ephemeral storage for data intensive workloads on Celery and Kubernetes workers. Previously, to accommodate larger workloads, you needed to integrate an external object storage or database tool to process large datasets within a single task. Now, you can customize the ephemeral storage when creating or updating your Deployment so that all data processing happens directly in your task Pod.

> You are only charged for used resources which are greater than the minimum defaults for each worker type:

Reviewer 2: ```suggestion
You are only charged for requested resources which are greater than the minimum defaults for each worker type:
```

## June 6, 2024

### API keys are no longer supported

As of June 1, 2024, Deployment API keys are no longer supported. Replace your API keys with [Deployment API tokens](https://www.astronomer.io/docs/astro/deployment-api-tokens), confirm successful operation with your API tokens, and then delete your API keys.

### Configure ephemeral storage on worker Pods

<HostedBadge/>

:::publicpreview
:::

You can now customize the amount of ephemeral storage for data intensive workloads on Celery, Kubernetes, and KubernetesPodOperator workers. Previously, to accommodate larger workloads, you needed to integrate an external object storage or database tool to process large datasets within a single task. Now, you can customize the ephemeral storage when creating or updating your Deployment so that all data processing happens directly in your task Pod.

You are only charged for requested resources which are greater than the minimum defaults for each worker type:

- **Celery worker**: 10 GiB minimum by default. 100 GiB maximum.
- **Kubernetes executor/ Kubernetes pod operator**: 0.25 GiB minimum by default. 100 GiB maximum.

> ### Astro Terraform provider

Author: ```suggestion
### Automate Airflow, resource, and infrastucture management with the Astro Terraform Provider
```

:::publicpreview
:::

> You can now use Terraform to automate setting up of organizations, workspaces, deployment, and teams with the Astronomer Terraform Provider package. The Terraform provider allows you to automate managing resources and changes to large organizations and Airflow infrastructure.

Author: ```suggestion
You can now use Terraform to automate managing resources and changes to large organizations and Airflow infrastructure with the Astronomer Terraform provider package. The provider is available through both the Terraform Registry and a public Github repository, where you can review the provider's code, make issues, and create pull requests.
```

> Refer to the [Astronomer Terraform Provider](https://registry.terraform.io/providers/astronomer/astro/latest/docs) docs for more information.

Author: ```suggestion
Refer to the Astronomer Terraform Provider  docs  in the [Terraform registry](https://registry.terraform.io/providers/astronomer/astro/latest/docs) or the [GitHub repository](https://github.com/astronomer/terraform-provider-astro) for more information.
```
